src/util/index_tree.ts

```diff
@@ -804,8 +806,14 @@ export class IndexTree<T extends IndexTreeNode<T>> {
         node.parent! as T,
         offset,
       );
-      node = node.parent!;
       path.push(sizeOfLeftSiblings + treePos.offset);
+      node = node.parent!;
+    } else if (node.hasTextChild()) {
+      const sizeOfLeftSiblings = addSizeOfLeftSiblings(
```

Comment:
If a node contains both TextNode and ElementNode as children, how should it be managed? 
Leaving a TODO comment for future reference would be good if this scenario is likely to arise.